app/master/slave.py

```diff
@@ -113,3 +113,32 @@ def free_executor(self):
         if new_count < 0:
             raise Exception('Cannot free executor on slave {}. All are free.'.format(self.url))
         return new_count
+
+    def is_alive(self, use_cached=True):
+        """
+        Is the slave API responsive?
+
+        :param use_cached: Should we use the last returned value of the network check to the slave? If True,
+            will return cached value. If False, this method will perform an actual network call to the slave.
+        :type use_cached: bool
+        :rtype: bool
+        """
+        if use_cached:
+            return self._is_alive
+
+        try:
+            response = self._network.get(self._slave_api.url())
+            self._is_alive = response.ok
+        except ConnectionError:
+            self._logger.warning('Slave with url {} is offline.', self.url)
+            self._is_alive = False
+
+        return self._is_alive
+
+    def set_is_alive(self, value):
```

Comment:
I was considering using the python properties (@property annotation) functionality for the getter and setter, but because the getter in this cause has to have an optional parameter, I decided to go through the old-school explicit route for the getter and setter.